shell/platform/android/io/flutter/embedding/engine/FlutterJNI.java

```diff
@@ -939,6 +942,117 @@ public void setLocalizationPlugin(@Nullable LocalizationPlugin localizationPlugi
 
   // ----- End Localization Support ----
 
+  // ----- Start Dynamic Features Support ----
+
+  /** Sets the dynamic feature manager that is used to download and install split features. */
+  @UiThread
+  public void setDynamicFeatureManager(@NonNull DynamicFeatureManager dynamicFeatureManager) {
+    ensureRunningOnMainThread();
+    this.dynamicFeatureManager = dynamicFeatureManager;
```

Comment:
I get the need for someone to hold this but given that this is a bridge, we should ideally not design this such that this is the "source of truth" for where the manager is. 

i.e. it relates to the customization question above but I think the solution should not be for users to set their preferred manager by calling this as the main front-facing API. The main manager customization API should be on the FlutterEngine or the DartExecutor (at which point we might ask whether the FlutterEngine or DartExecutor should just implement the manager interface but has-a manager implementation which it owns).

Reply:
For now, I would prefer to continue to do it this way, as it is consistent with the other similar use cases already in the JNI. We can do a refactor of the JNI, but that is a sizable task on its own.